### DISCUSSION TITLE
Single OIDC configuration for multiple vhosts

### DISCUSSION THREAD

**Discussion by cvuillemez** [OpenIDC/mod_auth_openidc#999] [General]
Hello,
For convenience, I'd like to share a single OIDC configuration for multiple virtual hosts in the same Apache Web Server instance.
But I wonder: is it a secure / recommended way or not ?

E.g., with the following macro than I could use for all vhosts:

```
 <Macro OIDC>
        OIDCSessionType client-cookie
        OIDCClientID myclient-id
        OIDCClientSecret XXXXXXXXXXXXXXXXXXXXXXXXXX
        OIDCProviderMetadataURL https://auth.example.com/.well-known/openid-configuration
        OIDCRedirectURI /redirect_uri
        OIDCCryptoPassphrase xxxxxxxxxxxx
 </Macro>
```

**Comment by zandbelt** [comment]
this is fine only if all of the vhosts on that server are in the same security domain i.e. under your control, used for the same purpose

**Reply by cvuillemez** [reply]
Thank you  :)
